lib/apiScenario/postmanCollectionRunnerClient.ts

```diff
@@ -693,19 +704,51 @@ pm.test("Stopped TestProxy recording", function() {
     baseUri: string,
     postScripts: string[],
     checkStatus: boolean = false,
-    responseAssertion?: StepResponseAssertion
+    responseAssertion?: StepResponseAssertion,
+    finalStateVia?: string,
+    isManagementPlane?: boolean
   ) {
     if (this.opts.skipLroPoll) return;
-
+    const url = item.request.url;
+    const urlStr = `${baseUri}${url.getPathWithQuery()}`;
     postScripts.push(
       `
-const pollingUrl = pm.response.headers.get("Location") || pm.response.headers.get("Azure-AsyncOperation") || pm.response.headers.get("Operation-Location");
+ const isArmResourceCreate = "${
+   isManagementPlane ?? false
+ }" === "true" && "${item.request.method.toLowerCase()}" === "put" && (pm.response.code === 201 || pm.response.code === 200)
```

Comment:
can we instead skip generating certain code for data-plane? for code readability